deps/parselibs.jl

```diff
@@ -0,0 +1,46 @@
+function execute(cmd::Cmd)
+    out = Pipe()
+    err = Pipe()
+  
+    process = run(pipeline(ignorestatus(cmd), stdout = out, stderr = err))
+    close(out.in)
+    close(err.in)
+  
+    (stdout = String(read(out)), 
+      stderr = String(read(err)),  
+      code = process.exitcode)
+end
+
+libparsepath = abspath(joinpath(@__DIR__, "..", "local", "bin", "libparse"))
+
+library_dir = ""
+
+if haskey(ENV, "SINGULAR_LIBRARY_DIR")
+    library_dir = ENV["SINGULAR_LIBRARY_DIR"]
+else
+    library_dir = abspath(joinpath(@__DIR__, "..", "local", "share", "singular", "LIB"))
+end
+
+filenames = filter(x -> endswith(x, ".lib"), readdir(library_dir))
+
+output_filename = abspath(joinpath(@__DIR__, "..", "src", "libraryfuncdictionary.jl"))
+
+open(output_filename, "w") do outputfile
+    println(outputfile, "libraryfunctiondictionary = Dict(")
+    for i in filenames
+        full_path = joinpath(library_dir, i)
+        libs = execute(`$libparsepath $full_path`)
+        if libs.stderr != ""
+            error("from libparse: $(libs.stderr)")
+        end
+        libs_splitted = split(libs.stdout,"\n")[4:end-1]
+        libs_splitted = [ split(i," ") for i in libs_splitted ]
```

Comment:
The next line becomes redundant if you do this instead:
```suggestion
        libs_splitted = [ split(i," ", keepempty = false) for i in libs_splitted ]
```

Also, it seems you prefer list comprehensions (which do me is very Pythonic), while I prefer classic FP style map/filter/... (probably because it is close to what I'd do in GAP). I have no idea what is considered to be "more typical Julia" or preferred in this project; perhaps it doesn't matter. But just to mention it, I think you could also do this (but this is not a change request):
```suggestion
        libs_splitted = map( i-> split(i," ", keepempty = false), libs_splitted ]
```